SAMtags.tex

```diff
@@ -234,13 +239,39 @@ \subsection{Barcodes}
 
 \begin{description}
 \item[BC:Z:\tagvalue{sequence}]
-Barcode sequence, with any quality scores stored in the {\tt QT} tag.
-
-\item[QT:Z:\tagvalue{qualities}]
-Phred quality of the barcode sequence in the {\tt BC} (or {\tt RT}) tag. Same encoding as {\sf QUAL}.
+Barcode sequence (Identifying the sample/library), with any quality scores (optionally) stored in the {\tt QT} tag.
+The {\tt BC} tag should match the {\tt QT} tag in length. 
+In the case of multiple unique molecular identifiers (e.g., one on each end of the template) the recommended implementation concatenates all the barcodes and qualities and places a dash (`{\tt -}') between the barcodes from the same template. 
```

Comment:
@yfarjoun I see your point that if you define the tag to be N called bases then if it's wrongly called you just chose a different N bases potentially missing part of the tag or incorporating bases beyond the tag. That way we correct for the indels via sequence alignment and assign the correct tag without changing length.  Good point!

Potentially we could have an instrument, eg some pyrosequencing method, that defined variable length tags in terms of a fixed number of "flows" (so AAA and A are both 1 flow), but this is hypothetical and even the 454 barcoding worked on fixed base count and not fixed flow count (I think).

So it's probably fine, although I'm curious what we gain by adding the restriction of being the same length.